packages/x-data-grid-pro/src/tests/dataSourceTreeData.DataGridPro.test.tsx

```diff
@@ -175,7 +175,8 @@ describeSkipIf(isJSDOM)('<DataGridPro /> - Data source tree data', () => {
 
     const firstChildId = (apiRef.current.state.rows.tree[GRID_ROOT_GROUP_ID] as GridGroupNode)
       .children[0];
-    apiRef.current?.dataSource.fetchRows(firstChildId);
+
+    await act(() => apiRef.current?.dataSource.fetchRows(firstChildId));
```

Comment:
Needs to receive a promise before `act` goes in async mode.
```suggestion
    await act(async () => {
      apiRef.current?.dataSource.fetchRows(firstChildId)
    });
```

Reply:
Wouldn't `await act(() => apiRef.current?.dataSource.fetchRows(firstChildId));` directly return a Promise `() => Promise<void>`, which is the the same as `async () => void`?

Reply:
oh, `apiRef.current?.dataSource.fetchRows(firstChildId)` returns a promise, ok didn't realize that. I'd still mark it explicitly as `async` since the `act` implementation depends on it and there is no other marker in this code (except for the types) that suggests it does.